File: data/cpp/tests/test_marshmallow_post_dump.cpp
#include <gtest/gtest.h>
#include "../src/function_marshmallow_post_dump.cpp"  

TEST(TestSerializeUserData, ValidUserData) {
    UserSchema schema;
    std::string user_data = R"({"age": 30, "score": 85.0})";
    std::string result = schema.serialize_user_data(user_data);
    EXPECT_NE(result.find("\"result\":"), std::string::npos);
    EXPECT_NE(result.find("\"age\":30"), std::string::npos);
    EXPECT_NE(result.find("\"score\":"), std::string::npos);
}

TEST(TestSerializeUserData, OutputStructure) {
    UserSchema schema;
    std::string user_data = R"({"age": 50, "score": 95.5})";
    std::string result = schema.serialize_user_data(user_data);
    EXPECT_NE(result.find("\"result\":"), std::string::npos);
}

TEST(TestSerializeUserData, ExtraFieldsInInput) {
    UserSchema schema;
    std::string user_data = R"({"age": 35, "score": 88.5, "extra": "field"})";
    std::string result = schema.serialize_user_data(user_data);
    EXPECT_NE(result.find("\"result\":"), std::string::npos);
    EXPECT_NE(result.find("\"age\":35"), std::string::npos);
    EXPECT_NE(result.find("\"score\":88.5"), std::string::npos);
    EXPECT_EQ(result.find("\"extra\":"), std::string::npos);
}

TEST(TestSerializeUserData, ZeroValues) {
    UserSchema schema;
    std::string user_data = R"({"age": 0, "score": 0.0})";
    std::string result = schema.serialize_user_data(user_data);
    EXPECT_NE(result.find("\"age\":0"), std::string::npos);
    EXPECT_NE(result.find("\"score\":0"), std::string::npos);
}

TEST(TestSerializeUserData, NegativeNumbers) {
    UserSchema schema;
    std::string user_data = R"({"age": -25, "score": -5.5})";
    std::string result = schema.serialize_user_data(user_data);
    EXPECT_NE(result.find("\"age\":-25"), std::string::npos);
    EXPECT_NE(result.find("\"score\":-5.5"), std::string::npos);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/java/tests/test_marshmallow_post_dump.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class test_marshmallow_post_dump {

    private function_marshmallow_post_dump schema;

    @BeforeEach
    void setUp() {
        schema = new function_marshmallow_post_dump();
    }

    @Test
    void test_valid_user_data() {
        String userData = "{\"age\": 30, \"score\": 85.0}";
        String result = schema.serialize_user_data(userData);
        String expected = "{\"result\":{\"age\":30,\"score\":85.0}}";
        assertEquals(expected, result, "Valid user data should be correctly serialized");
    }

    @Test
    void test_output_structure() {
        String userData = "{\"age\": 50, \"score\": 95.5}";
        String result = schema.serialize_user_data(userData);
        String expected = "{\"result\":{\"age\":50,\"score\":95.5}}";
        assertEquals(expected, result, "Output should have correct structure with result wrapper");
    }

    @Test
    void test_extra_fields_in_input() {
        String userData = "{\"age\": 35, \"score\": 88.5, \"extra\": \"field\"}";
        String expected = "{\"result\":{\"age\":35,\"score\":88.5}}";
        String result = schema.serialize_user_data(userData);
        assertEquals(expected, result, "Extra fields should be excluded from output");
    }

    @Test
    void test_zero_values() {
        String userData = "{\"age\": 0, \"score\": 0.0}";
        String expected = "{\"result\":{\"age\":0,\"score\":0.0}}";
        String result = schema.serialize_user_data(userData);
        assertEquals(expected, result, "Zero values should be handled correctly");
    }

    @Test
    void test_invalid_input() {
        String userData = "{\"age\": \"abc\", \"score\": \"def\"}";
        String result = schema.serialize_user_data(userData);
        assertEquals("Error: invalid input", result, "Invalid input should return error message");
    }
}